app/src/main/java/zapsolutions/zap/LandingActivity.java

```diff
@@ -27,6 +27,43 @@ protected void onCreate(Bundle savedInstanceState) {
         }
 
 
+        // support for clearing shared preferences, on breaking changes
+        if (PrefsUtil.getPrefs().contains(PrefsUtil.SETTINGS_VERSION)) {
+            int ver = PrefsUtil.getPrefs().getInt(PrefsUtil.SETTINGS_VERSION, RefConstants.CURRENT_SETTINGS_VERSION);
+            if (ver < RefConstants.CURRENT_SETTINGS_VERSION) {
```

Comment:
Should we maybe not use `current_version` but the fixed value like 17 otherwise we will reset the app next upgrade as well?